File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/implementation/mappers/EventsMapper.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.implementation.mappers;

import io.zenwave360.example.clinicaltool.common.mappers.BaseMapper;
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper(uses = {BaseMapper.class})
public interface EventsMapper {

    EventsMapper INSTANCE = Mappers.getMapper(EventsMapper.class);

    io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.PatientCreated asPatientCreated(
            Patient input);

    io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.DoctorCreated asDoctorCreated(Doctor input);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/implementation/mappers/EventsMapper.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.implementation.mappers

import io.zenwave360.example.clinicaltool.common.mappers.BaseMapper
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingTarget
import org.mapstruct.factory.Mappers


@Mapper(uses = [BaseMapper::class])
interface EventsMapper {

    companion object {
        val INSTANCE: EventsMapper = Mappers.getMapper(EventsMapper::class.java)
    }
    fun asPatientCreated(input: Patient): io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.PatientCreated
    fun asDoctorCreated(input: Doctor): io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.DoctorCreated
}
